engine/updategraph/src/main/java/io/deephaven/engine/updategraph/UpdateGraphProcessor.java

```diff
@@ -1314,27 +1322,35 @@ public ConcurrentNotificationProcessor(@NotNull final ThreadFactory threadFactor
         private void processSatisfiedNotifications() {
             log.info().append(Thread.currentThread().getName())
                     .append(": starting to poll for satisfied notifications");
-            while (running) {
-                Notification satisfiedNotification = null;
-                synchronized (satisfiedNotifications) {
-                    while (running && (satisfiedNotification = satisfiedNotifications.poll()) == null) {
-                        try {
-                            satisfiedNotifications.wait();
-                        } catch (InterruptedException ignored) {
+            Notification satisfiedNotification = null;
+            try {
+                while (running) {
+                    satisfiedNotification = null;
```

Comment:
This should be at the end of the loop, not the beginning, but still before the accounting.